### PR TITLE
Refactoring - Behaviour

### PR DESCRIPTION
**Pull request by hmerrettICHK**
**Description**
Refactoring of behaviour module - adding null coalesce

**Motivation and Context**

**How Has This Been Tested?**

**Screenshots**

### CODE REVIEW

File: modules/Behaviour/behaviour_manage_addMultiProcess.php [hunk 1]
```diff
@@ -41,18 +41,18 @@
...
+    $type = $_POST['type'] ?? '';
     $descriptor = null;
     if (isset($_POST['descriptor'])) {
         $descriptor = $_POST['descriptor'];
     }
     $level = null;
     if (isset($_POST['level'])) {
-        $level = $_POST['level'];
+        $level = $_POST['level'] ?? '';
     }
```
Comment (yookoala) [inline modules/Behaviour/behaviour_manage_addMultiProcess.php:53]: With null coalesce, there is no need to wrap the assignment in a if-then-isset check anymore. This is the same as:
```php
$level = $_POST['level'] ?? null;
```

File: modules/Behaviour/behaviour_manage_addMultiProcess.php [hunk 2]
```diff
@@ -41,18 +41,18 @@
...
         $gibbonPersonIDMulti = null;
     }
-    $date = $_POST['date'];
-    $type = $_POST['type'];
+    $date = $_POST['date'] ?? '';
+    $type = $_POST['type'] ?? '';
     $descriptor = null;
     if (isset($_POST['descriptor'])) {
         $descriptor = $_POST['descriptor'];
     }
```
Comment (yookoala) [inline modules/Behaviour/behaviour_manage_addMultiProcess.php:49]: The entire if-then check:
```php
    $descriptor = null;
    if (isset($_POST['descriptor'])) {
        $descriptor = $_POST['descriptor'];
    }
```
Is equivalent to this:

```php
$descriptor = $_POST['descriptor'] ?? null;
```

File: modules/Behaviour/behaviour_manage_editProcess.php [hunk 3]
```diff
@@ -68,23 +68,23 @@
...
-                $date = $_POST['date'];
-                $type = $_POST['type'];
+                $gibbonPersonID = $_POST['gibbonPersonID'] ?? '';
+                $date = $_POST['date'] ?? '';
+                $type = $_POST['type'] ?? '';
                 $descriptor = null;
                 if (isset($_POST['descriptor'])) {
-                    $descriptor = $_POST['descriptor'];
+                    $descriptor = $_POST['descriptor'] ?? '';
                 }
```
Comment (yookoala) [inline modules/Behaviour/behaviour_manage_editProcess.php:77]: Same as before, can be replaced with a simple null coalesce:

```php
$descriptor = $_POST['descriptor'] ?? null;
```

File: modules/Behaviour/behaviour_manage_addProcess.php [hunk 4]
```diff
@@ -176,11 +176,11 @@
...
         } elseif ($step == 2 and $gibbonBehaviourID != null) {
             //Proceed!
-            $gibbonPersonID = $_POST['gibbonPersonID'];
+            $gibbonPersonID = $_POST['gibbonPersonID'] ?? '';
             if ($_POST['gibbonPlannerEntryID'] == '') {
                 $gibbonPlannerEntryID = null;
             } else {
-                $gibbonPlannerEntryID = $_POST['gibbonPlannerEntryID'];
+                $gibbonPlannerEntryID = $_POST['gibbonPlannerEntryID'] ?? '';
             }
```
Comment (yookoala) [inline modules/Behaviour/behaviour_manage_addProcess.php:184]:
```php
            if ($_POST['gibbonPlannerEntryID'] == '') {
                $gibbonPlannerEntryID = null;
            } else {
                $gibbonPlannerEntryID = $_POST['gibbonPlannerEntryID'];
            }
```

Can be rewritten as:
```php
$gibbonPlannerEntryID = $_POST['gibbonPlannerEntryID'] ?? null;
```

File: modules/Behaviour/behaviour_manage_editProcess.php [hunk 5]
```diff
@@ -68,23 +68,23 @@
...
-                $comment = $_POST['comment'];
-                $followup = $_POST['followup'];
+                $comment = $_POST['comment'] ?? '';
+                $followup = $_POST['followup'] ?? '';
                 if ($_POST['gibbonPlannerEntryID'] == '') {
                     $gibbonPlannerEntryID = null;
                 } else {
-                    $gibbonPlannerEntryID = $_POST['gibbonPlannerEntryID'];
+                    $gibbonPlannerEntryID = $_POST['gibbonPlannerEntryID'] ?? '';
                 }
```
Comment (yookoala) [inline modules/Behaviour/behaviour_manage_editProcess.php:88]: Same as before:

```php
            if ($_POST['gibbonPlannerEntryID'] == '') {
                $gibbonPlannerEntryID = null;
            } else {
                $gibbonPlannerEntryID = $_POST['gibbonPlannerEntryID'];
            }
```

Can be rewritten as:
```php
$gibbonPlannerEntryID = $_POST['gibbonPlannerEntryID'] ?? null;
```

**Review by yookoala — CHANGES_REQUESTED** [review]
Please update the several changes mentioned.

File: modules/Behaviour/behaviour_manage_addMultiProcess.php (hunk 1 above)
Comment (hmerrettICHK) [reply modules/Behaviour/behaviour_manage_addMultiProcess.php:53]: Thanks for the feedback! I'm still quite new to refactoring, so I appreciate the advice. - Harry

**Review by SKuipers — CHANGES_REQUESTED** [review]
Hi Harry, can you follow up on these changes and click the 're-request review' when ready. Thanks!

**Pull request merged**